lib/middleware/parameter.js

```diff
@@ -140,10 +140,10 @@ module.exports = async (ctx, next) => {
                     const description = item.description || title;
                     const author = item.author || '';
                     let isFilter = true;
-                    ctx.query.filter_title && (isFilter = isFilter && !title.match(makeRegex(ctx.query.filter_title)));
-                    ctx.query.filter_description && (isFilter = isFilter && !description.match(makeRegex(ctx.query.filter_description)));
-                    ctx.query.filter_author && (isFilter = isFilter && !author.match(makeRegex(ctx.query.filter_author)));
-                    return !isFilter;
+                    ctx.query.filter_title && (isFilter = isFilter && title.match(makeRegex(ctx.query.filter_title)));
```

Comment:
```
 filter › filter filter_case_sensitive=false

    expect(received).toBe(expected) // Object.is equality

    Expected: 2
    Received: 0

      29 |         const response = await request.get('/test/1?filter=description4|title5&filter_case_sensitive=false');
      30 |         const parsed = await parser.parseString(response.text);
    > 31 |         expect(parsed.items.length).toBe(2);
         |                                     ^
      32 |         expect(parsed.items[0].title).toBe('Title4');
      33 |         expect(parsed.items[1].title).toBe('Title5');
      34 |     });

      at Object.toBe (test/middleware/parameter.js:31:37)
```
`filter`测试失败了，
因为`filter`参数是将`fitler_title`, `filter_description`, `filter_author`全部赋值`filter`，然后利用或运算过滤信息

你需要把`filter`参数单独拿出来进行过滤计算
`https://github.com/DIYgod/RSSHub/blob/master/lib/middleware/parameter.js`
134~138
```

if (ctx.query.filter) {
	ctx.state.data.item = ctx.state.data.item.filter((item) => {
		const title = item.title || '';
		const description = item.description || title;
		const author = item.author || '';
		const isFilter = title.match(makeRegex(ctx.query.filter_title)
						|| description.match(makeRegex(ctx.query.filter_description)
						|| author.match(makeRegex(ctx.query.filter_author);
		
		return isFilter !== null;
	});
}
// 启用filter参数时，无效filter_title/description/author
if (!ctx.query.filter && (ctx.query.filter_title || ctx.query.filter_description || ctx.query.filter_author)) {
	ctx.state.data.item = ctx.state.data.item.filter((item) => {
		const title = item.title || '';
		const description = item.description || title;
		const author = item.author || '';
		let isFilter = true;
		ctx.query.filter_title && (isFilter = isFilter && title.match(makeRegex(ctx.query.filter_title)));
		ctx.query.filter_description && (isFilter = isFilter && description.match(makeRegex(ctx.query.filter_description)));
		ctx.query.filter_author && (isFilter = isFilter && author.match(makeRegex(ctx.query.filter_author)));
		return isFilter !== null;
	});
}
```

Reply:
看不懂。还得修改哪里，老哥直接给例子吧，我直接复制修改，

Reply:
就是原来文件134~148行，替换成上述代码

Reply:
const isFilter = title.match(makeRegex(ctx.query.filter_title)
						|| description.match(makeRegex(ctx.query.filter_description)
						|| author.match(makeRegex(ctx.query.filter_author);
		```suggestion
                    ctx.query.filter_title && (isFilter = isFilter && title.match(makeRegex(ctx.query.filter_title)));
```

少了好几个右括号，唉，又折腾我好久。

Reply:
我拿记事本写的，没注意，不好意思